pkg/mermaid/endpointanalysisdiagram/endpointanalysisdiagram.go

```diff
@@ -0,0 +1,98 @@
+package endpointanalysisdiagram
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/anz-bank/sysl/pkg/sysl"
+)
+
+const projectDir = "../../../"
+
+//Keeps track of the statement-endpoint pairs we visit during execution
+type externalLink struct {
+	statement, endPoint string
+}
+
+//Accepts the sysl module and returns a string (and an error if any)
+//The resulting string is the mermaid code for the endpoint analysis for that application and endpoint
+func GenerateEndpointAnalysisDiagram(m *sysl.Module) (string, error) {
+	return generateEndpointAnalysisDiagram(m, &[]externalLink{}, true)
+}
+
+//This is a helper which has additional arguments which need not be entered by the user
+func generateEndpointAnalysisDiagram(m *sysl.Module, externalLinks *[]externalLink, theStart bool) (string, error) {
+	var result string
+	if theStart {
+		result = "%% AUTOGENERATED CODE -- DO NOT EDIT!\n\ngraph TD\n"
+	}
+	for appName, app := range m.Apps {
```

Comment:
Iterating over a map; does this not create non consistent mermaid code?
Maps don't ensure ordering, which is why sometimes you'll see this: https://github.com/joshcarp/sysl-printer/blob/master/alphabetical.go

Reply:
The order does not break the output or the code as long as all the apps are visited. Does the order of the output matter?